### PR TITLE
adjust FileSystem implementations to TSC API changes

### PR DESCRIPTION
**Pull request by tomerd**
motivation: TSC FileSystem API changed to add tempDirectory accessor

changes: add tempDirectory accessor in SwiftPM FileSystem implementations

rdar://86644116

### CODE REVIEW

**Comment by tomerd** [comment]
https://github.com/apple/swift-tools-support-core/pull/273

@swift-ci test this please

**Review by abertelrud — APPROVED** [review]

File: Sources/Basics/VirtualFileSystem.swift [hunk 1]
```diff
@@ -177,6 +177,8 @@ public class VirtualFileSystem: FileSystem {
 
     public var cachesDirectory: AbsolutePath? = nil
 
+    public var tempDirectory = AbsolutePath("/")
```
Comment (tomerd) [inline Sources/Basics/VirtualFileSystem.swift:180]: @neonichu doe that make sense in this case?

**Comment by tomerd** [comment]
apple/swift-tools-support-core#273

@swift-ci please smoke test

**Comment by tomerd** [comment]
apple/swift-tools-support-core#273

@swift-ci please smoke test

File: Sources/Commands/SwiftTool.swift [hunk 2]
```diff
@@ -1035,8 +1035,8 @@ private func getSharedSecurityDirectory(options: SwiftToolOptions, observability
 
     do {
         let sharedSecurityDirectory = try localFileSystem.getOrCreateSwiftPMSecurityDirectory()
-        // And make sure we can write files (locking the directory writes a lock file)
-        try localFileSystem.withLock(on: sharedSecurityDirectory, type: .exclusive) { }
+        // make sure we can write files
+        try withTemporaryFile(dir: sharedSecurityDirectory) { _ in }
```
Comment (tomerd) [inline Sources/Commands/SwiftTool.swift:1039]: @yim-lee fyi

**Comment by tomerd** [comment]
@swift-ci please test package compatibility

**Pull request merged**